src/client/Microsoft.Identity.Client/ManagedIdentity/AbstractManagedIdentity.cs

```diff
@@ -136,6 +141,36 @@ internal static string GetMessageFromErrorResponse(HttpResponse response)
             return $"[Managed Identity] Error Code: {managedIdentityErrorResponse.Error} Error Message: {managedIdentityErrorResponse.ErrorDescription}";
         }
 
+        // Try to get the error message from the nested error response in case of cloud shell.
+        private string TryGetMessageFromNestedErrorResponse(string response)
+        {
+            var json = JsonHelper.ParseIntoJsonObject(response);
+
+            JsonHelper.TryGetValue(json, "error", out var error);
+            JsonHelper.TryGetValue(JsonHelper.ToJsonObject(error), "message", out var message);
+            JsonHelper.TryGetValue(JsonHelper.ToJsonObject(error), "code", out var errorCode);
+
+            if (message == null && errorCode == null)
```

Comment:
you can return early with `errorMessage` and move this condition i.e. the generic error message to the bottom.